### DISCUSSION TITLE
Custom command to rename duplicate files with reference files names

### DISCUSSION THREAD

**Discussion by AlexMichaan** [hsoft/dupeguru#1004] [Q&A]
Hi! 
I am looking for a way to write a custom command that would rename the duplicate selected file with the name of the original reference file above. Unfortunately I am not at all familiar with the custom command tool in Dupeguru and I didn't manage to write a proper one to fill that purpose. 
The goal is to gain time by being able to do this renaming action in a single click, instead of having to copy-paste manually the reference file's name onto the duplicate one. It would help me immensely since I am currently using Dupeguru to work on a very large amount of duplicate pictures that need renaming. 
If ever you have a tip for a custom command that would work, I would be very thankful. 
I work on mac OS 10.12 with Dupeguru 4.0.3. 
Many thanks,
Alex

**Comment by glubsy** [comment]
You could create a script like this:
```zsh
#!/bin/zsh
mv "$1" "$1_dupeguru_temp_rename"
mv "$2" "$1"
mv "$1_dupeguru_temp_rename" "$2"
```
and have this as the custom command: `"/path/to/my/script.sh" """%r""" """%d"""` (triple double-quotes to avoid parsing issues if quotes are already in  filename)

This is a legitimate feature request, and I have had this on my TODO list for a while, maybe I'll get around to implement it in the future.

**Reply by AlexMichaan** [reply]
Thanks a lot for the answer! Unfortunately I cannot manage to make it work somehow. I must be doing some step wrong since I am not familiar with scripts. I have created the script you advised, using textedit in plain text format and saved it in unicode with .sh extension, and indicated the correct path in the custom command following your instructions. For now nothing happens when I run the costum command on a selected duplicate. 
Sorry if the step I am missing is an obvious one. If ever you have any idea of what I didn't do correctly, I'll be thankful

**Reply by glubsy** [reply]
I suggest debugging your script step by step by adding something like `echo "This step has run" > ~/dupeguru_test_log.txt` after each line (this will print out to the file in your home directory, to make sure the script is actually invoked and run by the shell)

**Reply by AlexMichaan** [reply]
Ok thank you, I will try this to see if the script in running

**Comment by rkazak** [comment]
Did you resolve this?

**Reply by Ikkoru** [reply]
I'm pretty sure the suggested script is just wrong. I think it swaps the two files places.
I'm currently looking for a way to append one filename to another. If I figure that out, I'll write here.

EDIT: Figured it out~

**Comment by Ikkoru** [comment]
I'm not a scripter, but I managed to find and edit some options for Custom Command renaming:
- Rename the **duplicate** to be the same as the **reference** (have **only** the duplicates selected):
`powershell Rename-Item -LiteralPath '%d' -NewName ([System.IO.Path]::GetFileName('%r'))`

- Rename the **reference** to be the same as the **duplicate** (have **only** the duplicates selected):
`powershell Rename-Item -LiteralPath '%r' -NewName ([System.IO.Path]::GetFileName('%d'))`

- Append the **reference** to the **duplicate** (have **only** the duplicates selected):
`powershell -Command "Rename-Item -LiteralPath '%r' -NewName (([System.IO.Path]::GetFileNameWithoutExtension('%r')).ToString() + ' ' + ([System.IO.Path]::GetFileNameWithoutExtension('%d')).ToString() + ([System.IO.Path]::GetExtension('%r')).ToString())"`

- Append the **duplicate** to the **reference** (have **only** the duplicates selected):
`powershell -Command "Rename-Item -LiteralPath '%d' -NewName (([System.IO.Path]::GetFileNameWithoutExtension('%d')).ToString() + ' ' + ([System.IO.Path]::GetFileNameWithoutExtension('%r')).ToString() + ([System.IO.Path]::GetExtension('%d')).ToString())"`
